### PR TITLE
added mergeAllProps

### PR DESCRIPTION
**Pull request by BjornMelgaard**
discussion: https://github.com/ramda/ramda/issues/2295#issuecomment-331532072

### CODE REVIEW

**Comment by BjornMelgaard** [comment]
Cant fix codeclimate, he swears for no reason

**Comment by char0n** [comment]
@BjornMelgaard thanks for PR. I will do some cleanup before merging. Also I am inclined to rename the function to `mergeProps`, because `mergeAllProps` is IMHO not very correct. It doesn't merge all the props but only props specifically listed..thus `mergeProps`.

Also in future I will be adding `mergePaths`. ramda-adjuncts contains also `paths` which we can use composing `mergePaths`.

File: src/index.d.ts [hunk 1]
```diff
@@ -344,6 +344,12 @@ declare namespace RamdaAdjunct {
         resetToDefault(defaultOptions: Object): (options: Object) => Object; // alias of mergeRight
 
         /**
+         * Functional equivalent of merging object properties with object spread.
+         */
+        mergeAllProps(ps: string[], obj: Object): Object; // alias of mergeRight
```
Comment (char0n) [inline src/index.d.ts]: remove alias comment

File: src/index.d.ts [hunk 2]
```diff
@@ -344,6 +344,12 @@ declare namespace RamdaAdjunct {
...
+        mergeAllProps(ps: string[]): (obj: Object) => Object; // alias of mergeRight
```
Comment (char0n) [inline src/index.d.ts]: remove alias comment

File: src/mergeAllProps.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * const obj = {
+ *   foo: { fooinner: 1 },
+ *   bar: { barinner: 2 }
+ * };
+ * // this statements are equal
+ * const expected   = { fooinner: 1, barinner: 2 };
+ * const withSpread = { ...obj.foo, ...obj.bar }
+ * const withFunc   = R.mergeAllProps(['foo', 'bar'], obj)
+ */
+const mergeAllProps = curry((ps, object) => mergeAll(props(ps, object)));
```
Comment (char0n) [inline src/mergeAllProps.js]: rename to mergeProps

File: src/mergeAllProps.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { curry, mergeAll, props } from 'ramda';
+
+/**
+ * Functional equivalent of merging object properties with object spread.
+ *
+ * @func
+ * @memberOf RA
+ * @since {@link TODO}
```
Comment (char0n) [inline src/mergeAllProps.js]: add real version

File: src/mergeAllProps.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * @category Object
+ * @sig [k] -> {k: {a}} -> {a}
+ * @param {!Array} ps The property names to fetch
+ * @param {!Object} obj The object to query
+ * @return {!Object} A merged properties of object.
```
Comment (char0n) [inline src/mergeAllProps.js]: remove ending dot

**Comment by BjornMelgaard** [comment]
done

**Comment by char0n** [comment]
@BjornMelgaard rebased your work here: https://github.com/char0n/ramda-adjunct/pull/135

Do a code review, add what you feel should be added, but use [conventional changelogg](https://github.com/char0n/ramda-adjunct/blob/7b1721c9e4a0a0112f3320e6fd18edbd004ec336/CONTRIBUTING.md#type) for commit message style.

We'll close this one

**Comment by BjornMelgaard** [comment]
ok

**Pull request closed without merging**